Add better data files presentation to isa-tab datatype

djcomlab opened the pull request:
Add better data files presentation to `isa-tab` datatype. Displays each type of data files in the HTML preview/presentation separately, and as collapsable sections.

mvdbeek committed:
Allow Python 3 if in a virtualenv

mvdbeek committed:
Move mercurial imports inside the functions that use them

Because Galaxy under Python3 would fail otherwise.

pkrog committed:
Add grooming for composite files.

Add call to groom_dataset_content() for composite files.

pkrog committed:
Add grooming for composite files.

Add call to groom_dataset_content() for composite files.

pkrog committed:
Add ISA composite data type.

Add new data types isa-tab and isa-json.
Add isa-rwval library (isatools Python 2 version) in requirements.

pkrog committed:
Add ISA composite data type.

Add new data types isa-tab and isa-json.
Add isa-rwval library (isatools Python 2 version) in requirements.

pcm32 committed:
[ISA datatype] adds logger setting for isatools.isatab to avoid failing uploads when no pandas (#31)

* [ISA datatype] adds logger setting for isatools.isatab to avoid failing uploads when no pandas.

* Update isa.py

pcm32 committed:
[ISA datatype] adds logger setting for isatools.isatab to avoid failing uploads when no pandas (#31)

* [ISA datatype] adds logger setting for isatools.isatab to avoid failing uploads when no pandas.

* Update isa.py

Commit:
Add a condition to FilterFailedDatasetsTool to remove empty files

Hello,

Here is an extra-condition for the tool ``FilterFailedDatasetsTool`` to allow it to remove empty datasets as well as failed datasets in a collection. I tested it on my data and it seems to work fine. Since it's my first time looking at galaxy's source code, I don't know if there is a proper way to do that ... And I didn't make the improvement for paired datasets because I don't have proper data to test it. But I assume it would be something like ``if forward_o.is_ok and reverse_o.is_ok and forward_o.has_data() and reverse_o.has_data():`` ?

Commit:
improved filtering of empty files in a collection (1)

- Supplemental parameter : 'remove_empty' which allow the user to choose if he/she wants to keep or not empty files in the collection.
- Coming next : the modified wrapper.
(Edited directly on GitHub).

Commit:
modified wrapper filter_failed_collection

- Supplemental 'select' parameter : 'remove_empty'

pcm32 committed:
Installs isa-rwval from develop

Otherwise failing currently when building in the container.

pcm32 committed:
Installs isa-rwval from develop

Otherwise failing currently when building in the container.

pkrog committed:
Correct linting issues detected by Travis-CI run.

pkrog committed:
Correct linting issues detected by Travis-CI run.

pkrog committed:
Correct again lint errors for Travis-CI.

pkrog committed:
Correct again lint errors for Travis-CI.

pkrog committed:
Correct some more lint errors for Travis-CI.

pkrog committed:
Correct some more lint errors for Travis-CI.

pkrog committed:
Change ISA Tools web site link.

Replace ISA Tools .org web site that has just been hacked, by the GitHub
page of the organization.

pkrog committed:
Change ISA Tools web site link.

Replace ISA Tools .org web site that has just been hacked, by the GitHub
page of the organization.

pkrog committed:
Escape filename in primary file.

pkrog committed:
Escape filename in primary file.

pkrog committed:
Escape assay information in ISA display_data().

pkrog committed:
Escape assay information in ISA display_data().

pkrog committed:
Remove unused method _list_archive_files().

pkrog committed:
Remove unused method _list_archive_files().

pkrog committed:
Use CompressedFile class.

Replace unzip custom code by Galaxy call to CompressedFile class.

pkrog committed:
Use CompressedFile class.

Replace unzip custom code by Galaxy call to CompressedFile class.

pkrog committed:
Remove logger configuration.

pkrog committed:
Remove logger configuration.

pkrog committed:
Comment out raw files listing in ISA type.

pkrog committed:
Comment out raw files listing in ISA type.

pkrog committed:
Remove dynamical setting of study name for ISA.

pkrog committed:
Remove dynamical setting of study name for ISA.

machinelit committed:
Add new lazy table parsing for isa-tab datatype and reintroduce data files

machinelit committed:
Add new lazy table parsing for isa-tab datatype and reintroduce data files

pkrog committed:
Start writing upload test for isa-tab.

Do not know how to read zip file content and submit it to uploader for a
composite type.

pkrog committed:
Start writing upload test for isa-tab.

Do not know how to read zip file content and submit it to uploader for a
composite type.

machinelit committed:
Introduce new table parsers to get data files back; temporarily puts dependency back to Git repo to test against (updated PyPI package for isa-rwval will be pushed soon).

machinelit committed:
Introduce new table parsers to get data files back; temporarily puts dependency back to Git repo to test against (updated PyPI package for isa-rwval will be pushed soon).

machinelit committed:
Fix requirements again for isa-rwval==0.10.1

machinelit committed:
Fix requirements again for isa-rwval==0.10.1

machinelit committed:
Cleanup presentation of data files (unused rendering using datafile id)

machinelit committed:
Cleanup presentation of data files (unused rendering using datafile id)

dannon committed:
Backbone model url fixes (urlRoot isn't necessary in many of these situations, and only complicates (and breaks) some things

Convert history-contents urlroot to instance var

remove outdated comment

jmchilton committed:
Fix file type error in RuleCollectionBuilder.

jmchilton committed:
Fix collection mapping to allow mapping over multi-data inputs.

It is jarring to users that works for collection inputs but not multiple data inputs, especially because the IUC discourages collection inputs when multiple data inputs would work.

mvdbeek committed:
Fix up subcollection_type

mvdbeek committed:
Simplify and modularize FilterFailedDatasetsTool class

mvdbeek committed:
Fix can_map_over for mapping over paired inputs

We'd only try mapping over the first collection type description
otherwise. I suppose we didn't notice because the first type is `list`
and therefore much more common. Noticed this while working on https://github.com/galaxyproject/galaxy/pull/5640.

mvdbeek committed:
Fix can_map_over for mapping over paired inputs

We'd only try mapping over the first collection type description
otherwise. I suppose we didn't notice because the first type is `list`
and therefore much more common. Noticed this while working on https://github.com/galaxyproject/galaxy/pull/5640.

nsoranzo committed:
Move `get_mercurial_default_options_dict()` where it's used

scrathat committed:
Fix tool search overlay link

mvdbeek committed:
Fix inferring output structure when mapping over

Previously this would have always generated `list:list` structures,
even when the input is a list:pair. I'm not 100% convinced
this is correct, but this seems to work.

mvdbeek committed:
Fix inferring output structure when mapping over

Previously this would have always generated `list:list` structures,
even when the input is a list:pair. I'm not 100% convinced
this is correct, but this seems to work.

nsoranzo committed:
Use Mercurial CLI instead of API for `add_changeset()`

mvdbeek committed:
Add filter empty collection tool

I think having a separate tool for this task makes sense and makes the
distinction between empty and fialed datasets clearer.

nsoranzo committed:
Use Mercurial CLI instead of API for `commit_changeset()`

jmchilton committed:
Fix delete "Collection Only" option in history menu.

Broken with https://github.com/galaxyproject/galaxy/commit/7cc01380afd017c5cac2af2b2d735b2265774a68.

nsoranzo committed:
Use Mercurial CLI instead of API for `pull_repository()`

Merge `get_configured_ui()` into `get_repo_for_repository()`.

nsoranzo committed:
Use Mercurial CLI instead of API for `remove_file()`

nsoranzo committed:
Remove unnecessary calls to `get_repo_for_repository()`

mvdbeek committed:
Break test by adding "list" to collection_type

This will break tests trying to map over list:pair collections,
for instance test_extract_workflows_with_dataset_collections.

mvdbeek committed:
Break test by adding "list" to collection_type

This will break tests trying to map over list:pair collections,
for instance test_extract_workflows_with_dataset_collections.

guerler committed:
Sort history items by history id in success message

jmchilton committed:
Merge pull request #6289 from guerler/fix_update_handler

Only update options if provided in parameter definition

guerler committed:
Hide inputs for tools without inputs

mvdbeek committed:
Add test tool and workflow for mapping over with mapping output

This doesn't quite work because of the dynamic output collection,
which fails with:

```
galaxy.tools.parameters.output_collect DEBUG 2018-06-06 18:40:38,784 (3) Add dynamic collection datasets to history for output [reverse] (171.970 ms)
galaxy.tools.parameters.output_collect ERROR 2018-06-06 18:40:38,839 Problem gathering output collection.
Traceback (most recent call last):
  File "/Users/mvandenb/src/galaxy/lib/galaxy/tools/parameters/output_collect.py", line 340, in collect_dynamic_outputs
    collection_builder.populate()
  File "/Users/mvandenb/src/galaxy/lib/galaxy/dataset_collections/builder.py", line 88, in populate
    elements = self.build_elements()
  File "/Users/mvandenb/src/galaxy/lib/galaxy/dataset_collections/builder.py", line 59, in build_elements
    new_elements[identifier] = element.build()
AttributeError: 'HistoryDatasetAssociation' object has no attribute 'build'
```

This seems to happen because the inner collection is wrongly detected as
nested. In general I doubt that mapping over colleciton output works
with dynamically discovered output collections.

mvdbeek committed:
Add test tool and workflow for mapping over with mapping output

This doesn't quite work because of the dynamic output collection,
which fails with:

```
galaxy.tools.parameters.output_collect DEBUG 2018-06-06 18:40:38,784 (3) Add dynamic collection datasets to history for output [reverse] (171.970 ms)
galaxy.tools.parameters.output_collect ERROR 2018-06-06 18:40:38,839 Problem gathering output collection.
Traceback (most recent call last):
  File "/Users/mvandenb/src/galaxy/lib/galaxy/tools/parameters/output_collect.py", line 340, in collect_dynamic_outputs
    collection_builder.populate()
  File "/Users/mvandenb/src/galaxy/lib/galaxy/dataset_collections/builder.py", line 88, in populate
    elements = self.build_elements()
  File "/Users/mvandenb/src/galaxy/lib/galaxy/dataset_collections/builder.py", line 59, in build_elements
    new_elements[identifier] = element.build()
AttributeError: 'HistoryDatasetAssociation' object has no attribute 'build'
```

This seems to happen because the inner collection is wrongly detected as
nested. In general I doubt that mapping over colleciton output works
with dynamically discovered output collections.

nsoranzo committed:
Fix `GALAXY_RUN_ALL=1 ./run.sh --daemon`

Broken in commit 0ea8107191ea1a3bc8b4b57623c981c06f38092e .
Reported by @lparsons in
https://github.com/galaxyproject/galaxy/pull/6262#issuecomment-395113621

natefoo committed:
Merge pull request #6295 from nsoranzo/release_18.01_followup_on_6262

[18.01] Fix `GALAXY_RUN_ALL=1 ./run.sh --daemon`

martenson committed:
Merge pull request #6274 from guerler/fix_help

 Fix help text for renaming operation in workflows

martenson committed:
Merge pull request #6281 from jmchilton/fix_delete_collection_only_2

Fix delete "Collection Only" option in history menu.

martenson committed:
Merge pull request #6266 from jmchilton/from_param_fixes_backport

[18.01] Backport fixes for collection output actions.

martenson committed:
Merge pull request #6286 from guerler/column_performance

Index select2 field values

guerler committed:
Merge pull request #6240 from dannon/workflow_submit_text_links

Workflow submit text links

nsoranzo committed:
Merge branch 'release_18.01' into release_18.05

nsoranzo committed:
Merge branch 'release_18.05' into dev

nsoranzo committed:
Use `hg init` to create a Mercurial repository

This allows removing a call to `hg_util.get_repo_for_repository()`, which
uses the discouraged Mercurial API.

pkrog committed:
Pinned requirements as a symbolic link as done in dev branch.

pkrog committed:
Pinned requirements as a symbolic link as done in dev branch.

machinelit committed:
Add check so that empty filenames not added to file list.

machinelit committed:
Add check so that empty filenames not added to file list.

mvdbeek committed:
Test list collection mapping over multidata input

mvdbeek committed:
Test list collection mapping over multidata input

mvdbeek committed:
Fix mapping list over multidata input

mvdbeek committed:
Fix mapping list over multidata input

martenson committed:
add the canonical url to grt config

martenson committed:
Merge pull request #6270 from galaxyproject/fix_workflow_link

fix link to workflow list

nsoranzo committed:
Remove unused functions

davebx committed:
Add specific datatype for CuffDiff SQLite database.

jmchilton committed:
Merge pull request #6300 from mvdbeek/multidata_list_input

Make simple lists selectable again in multidata inputs

jmchilton committed:
Merge pull request #6277 from nsoranzo/mercurial_api_removal2

Consolidate use of get_repo_for_repository() and clone_repository()

nsoranzo committed:
Use `hg id` to map a changeset to a revision number

This allows removing a lot of calls to `hg_util.get_repo_for_repository()`,
which uses the discouraged Mercurial API.

nsoranzo committed:
Fix `reversed_upper_bounded_changelog()` calls

`hg_util.reversed_upper_bounded_changelog()` expects a changeset revision
string as second parameter, not a `changectx` object.

martenson committed:
Merge pull request #6301 from martenson/grt-config

[18.05] add the canonical url to grt config

mvdbeek committed:
Merge pull request #6076 from dannon/workflow_fixes

Workflow and remote-user related refactoring and a bug fix

mvdbeek committed:
Increase sleep if encountering exception

rhpvorderman committed:
Fix default job script by moving env setup

The part where pythonpath was evaluated totally ignored any variables set in the env setup. That is now fixed. All credits go to @mvdbeek for finding the cause of the issue and proposing the solution.

rhpvorderman committed:
Change location of env setup

nsoranzo committed:
Merge pull request #6304 from mvdbeek/sleep_more

[18.05] Increase sleep if encountering (DB locking) exception

nsoranzo committed:
Put import at the top

machinelit committed:
Implement presentation enhancements to ISA-Tab datatype for listing data files

machinelit committed:
Implement presentation enhancements to ISA-Tab datatype for listing data files

davebx committed:
Properly store metadata for cuffdiff.sqlite datatype.

martenson committed:
Merge pull request #6238 from dannon/1805stylefixes_rebased

[18.05] Various style fixes for history, toolshed, messages (backport)

martenson committed:
Merge pull request #6243 from dannon/workflow_submit_text_backport

[18.05] Backport of Workflow Execution Text Fixup

martenson committed:
Merge pull request #6226 from dannon/minor_style_fixes

[18.05] Two minor aliases for (exterior) backwards compatibility, minor badge tweaks

martenson committed:
remove outdated comment

martenson committed:
Merge pull request #6237 from dannon/history_contents_multiview

[18.05] Fix history contents w/ proxy prefix (multiview error)

martenson committed:
Merge pull request #6260 from dannon/history-view-fixes

[18.05] History view 'import' button display based on ownership.

martenson committed:
backport and build client

jmchilton committed:
Merge pull request #6311 from martenson/backport-and-build

[18.05] Backport and build client changes

mvdbeek committed:
Need to set default HOME/TMP before env_setup_commands

Otherwise the defaults take precedence

rhpvorderman committed:
Merge pull request #1 from mvdbeek/rhpvorderman-patch-1

Need to set default HOME/TMP before env_setup_commands

mvdbeek committed:
Consider mapping-over when picking collection_type for dynamically discovered collections

mvdbeek committed:
Consider mapping-over when picking collection_type for dynamically discovered collections

dannon committed:
Merge pull request #6307 from nsoranzo/mercurial_api_removal3

More use of Mercurial CLI instead of API

dannon committed:
Merge pull request #6292 from hrhotz/patch-1

Update tabular.py

dannon committed:
Missing semicolon.

dannon committed:
Client format

dannon committed:
Merge branch 'workflow.tool.search.clear' into dev

dannon committed:
Client format (from the merge)

dannon committed:
Merge pull request #6305 from rhpvorderman/patch-1

Fix default job script by moving env setup

dannon committed:
Remove stray doublequote (which was showing up in the link)

dannon committed:
Merge pull request #6142 from scrathat/workflow_toollink_area

Make workflow editor toolTitle link include the tools description

natefoo committed:
Merge pull request #6293 from nsoranzo/release_18.05_strip_slurm_warning

[18.05] Strip a spurious Slurm warning from job stderr

dannon committed:
Missing semicolon.

nsoranzo committed:
Merge branch 'release_18.05' into dev

dannon committed:
Client format

dannon committed:
Merge branch 'job_success_message' into dev

machinelit committed:
Add number of files per file type

machinelit committed:
Add number of files per file type

mvdbeek committed:
Recover the correct DataCollectionToolParameter when checking if a collection is being mapped over

mvdbeek committed:
Recover the correct DataCollectionToolParameter when checking if a collection is being mapped over

nsoranzo committed:
Replace unmaintained pycrypto dependency with pycryptodome

Minimum secret key length for Blowfish was increased from 4 to 5 bytes.

First argument to Blowfish.new() must be a bytestring, and the `mode`
argument is no more optional.

Remove unused `encode_id` function from `scripts/api/common.py` .

Documentation updated, then rebuilt with `make config-rebuild` .

jmchilton committed:
Merge pull request #6278 from mvdbeek/fix_paired_mapping_over

Various fixes for mapping over collections

mvdbeek committed:
Fix typo (UnitializedTree -> UninitializedTree)

nsoranzo committed:
Add py34-first_startup to Travis and default tox tests.

mvdbeek committed:
Merge pull request #6066 from jmchilton/apply_rules_tutorials

Rules - test case for apply rules tutorial.

mvdbeek committed:
Fix undefined success_message

jmchilton committed:
Merge pull request #6315 from mvdbeek/fix_uninitialized_typo

Fix typo (UnitializedTree -> UninitializedTree)

mvdbeek committed:
Allow #if $datasets #end if patern for file lists

Otherwise the following exception occurs:
```
Traceback (most recent call last):
  File "/home/wolffj/src/galaxy/lib/galaxy/jobs/runners/__init__.py", line 191, in prepare_job
    job_wrapper.prepare()
  File "/home/wolffj/src/galaxy/lib/galaxy/jobs/__init__.py", line 858, in prepare
    self.command_line, self.extra_filenames, self.environment_variables = tool_evaluator.build()
  File "/home/wolffj/src/galaxy/lib/galaxy/tools/evaluation.py", line 435, in build
    raise e
AttributeError: 'DatasetListWrapper' object has no attribute 'value'
```

This is because DatasetListWrapper inherits `__bool__` from ToolParameterValueWrapper
which defines bool as
```
    def __bool__(self):
        return bool(self.value)
    __nonzero__ = __bool__
```
while DatasetListWrapper has no `value` attribute.

Fixes https://github.com/galaxyproject/galaxy/issues/6314 reported by
@joachimwolff (many thanks for the detailed report!).

mvdbeek committed:
Use #if $datasets #end if pattern in test tool

nsoranzo committed:
Update dependencies to versions with Python3 manylinux wheels on PyPI

dannon committed:
Merge pull request #6316 from mvdbeek/success_message_fix

Fix undefined success_message

mvdbeek committed:
Fix __bool__ for empty optional multidata parameters

mvdbeek committed:
Add a test for not providing an optional input

hexylena committed:
Merge pull request #6317 from mvdbeek/fix_6314

Allow #if $datasets #end if pattern for file lists

mvdbeek committed:
Merge pull request #6269 from nsoranzo/phenotype_association_tools

Fixes for phenotype association tools

mvdbeek committed:
Merge pull request #6176 from dannon/user_grid_impersonate

Add impersonate to main user grid when it's enabled

martenson committed:
Merge pull request #6148 from jmchilton/cwl-small-stuff-2

Small fixes and small refactorings toward reuse in CWL branch (retry)

nsoranzo committed:
Python3 Unicode/binary fixes

scrathat committed:
Fix target update for rule collection builder after input selection

mvdbeek committed:
Merge pull request #6326 from scrathat/rule.builder.selection.update

Fix target update for rule collection builder after input selection

jmchilton committed:
Merge pull request #6325 from mvdbeek/py3_fixes2

 Add py34-first_startup to Travis and default tox tests

mvdbeek committed:
Change get_new_elements to _get_new_elements

And use same order for arguments in signature and caller.

mvdbeek committed:
Correct help text of filter_failed_collection and filter_empty_collection tools

mvdbeek committed:
Add tests for filtering list:pair and list:list:pair collections

mvdbeek committed:
Map over list or list:paired collection in filter collection tools

dannon committed:
Merge pull request #6303 from guerler/history_polling

Stop listening to history changes after job submission

mvdbeek committed:
Update comments to the fact that we're processing list or list:paired collections

dannon committed:
Rebuild client w/ maps.

bernt-matthias committed:
added missing level fatal_oom to xsd

bernt-matthias committed:
also added missing documentation

dannon committed:
Initial python-based plugin staging.  to be used from common startup.

dannon committed:
Call plugin static dir regeneration in the event client isn't being built (which also does this, albeit slightly differently)

mvdbeek committed:
Update description for installing pbs-python

dannon committed:
Minor refactoring, comments.

dannon committed:
Add static-safe mapping for uwsgi for plugin static dirs, which are accessible via symlink

dannon committed:
Use os.pardir instead of ../.  Thanks @nsoranzo

nsoranzo committed:
Merge pull request #6345 from mvdbeek/update_pbs_python_doc

[18.01] Update description for installing pbs-python

nsoranzo committed:
Small fixes

dpryan79 committed:
Prevent throwing an exception if a dataset doesn't exist (and never will) due to a job error. This is an alternative (and better) solution to #6337 when jobs set metadata externally.

martenson committed:
Merge pull request #6344 from dannon/python_static_staging

[18.05] Visualization static staging

jmchilton committed:
Merge pull request #6349 from dpryan79/preventExceptionOnFileAccess

Prevent throwing an exception if a dataset doesn't exist (and never w…

jmchilton committed:
Merge pull request #6341 from bernt-matthias/release_18.05_fix_xsd

[18.05] Added missing level fatal_oom to xsd

mvdbeek committed:
Merge pull request #5640 from Mataivic/patch-1

Add a FilterEmptyDatasetsTool and allow filtering tools to apply to arbitrary collection structures

bgruening committed:
Merge pull request #6336 from dannon/backport_url_fix_flattened

[18.01] Backport backbone url fixes (flattened)

nsoranzo committed:
Merge branch 'release_18.01' into release_18.05

nsoranzo committed:
Merge branch 'release_18.05' into dev

bgruening committed:
Merge pull request #6302 from davebx/cuffdiff_datatype

Add specific datatype for CuffDiff SQLite database.

jmchilton committed:
Merge pull request #6234 from mvdbeek/fix_relabel_non_strict

[18.01] Fix non-strict tabular relabel operation

jmchilton committed:
Merge pull request #6072 from mvdbeek/add_resume_job_api

Add resume job api

martenson committed:
grt - wrap every write in try-except

 and just fail the row if the data is crap
(like {NULL, SOH, STX, ETX} and friends)

martenson committed:
unname the exception

dannon committed:
Merge pull request #6354 from martenson/superdefensive-grt

[18.05] grt - wrap every write in try-except

dannon committed:
Fix share with another user in 18.05

jmchilton committed:
Merge pull request #6358 from dannon/1805sharingfix

[18.01] Fix 'share with another user'

guerler committed:
Fix size handling in legacy modal view

guerler committed:
Limit modal body size to 67 percent of window height

guerler committed:
Add fasta datatype to msa viewer

guerler committed:
Add phylip for msa viewer

dannon committed:
Merge pull request #6361 from guerler/fix_msa

[18.05] Add fasta datatype to msa viewer

dannon committed:
Merge pull request #6360 from guerler/legacy_modal_scrolling

[18.05] Fix size handling in legacy modal view

dannon committed:
Merge remote-tracking branch 'jmchilton/collection_fixes' into release_18.05

dannon committed:
Client build.

dannon committed:
Merge remote-tracking branch 'upstream/release_18.05' into dev

jmchilton committed:
Optimizations for dynamic output files collection.

During the set hid and update quota parts of this operation - skip a bunch of extra flushes and skip checking for extra files since these cannot be created in conjunction with dynamic file discovery currently.

Cut this part of that operation from about 50 seconds to 7 when creating a collection with 1000 elements on my laptop against a local postgres database.

jmchilton committed:
Fix logging bug.

nsoranzo committed:
Merge branch 'release_18.01' into release_18.05

nsoranzo committed:
Merge branch 'release_18.05' into dev

jmchilton committed:
More correct, more efficient collection operations.

We were copying datasets and not assigning them new HIDs in some cases (the newer collection operations apply_rules, tag_from_file were doing the right thing here but it looks like we didn't go back and fix older operations).

With the uniform handling of dataset copying in place - I used the much more efficient history.add_datasets instead of history.add_dataset.

This also optimizes the actual HDA copying by eliminating some extra flushes. hda.copy() would flush to get an HDA before copying metadata but this is only needed if there are FileParameter metadata types (e.g. bams) so I made that flush conditional as well as the final one. So for most datatypes - this eliminates two extra flushes per dataset copied.

Rebased with fix for issue pointed out by @nsoranzo - cleaned up _add_datasets_to_history to just consume values directly. Using itervalues since these methods have performance issues and these lists can be quite large.

jmchilton committed:
Remove debug statement when creating certain kinds of collections.

martenson committed:
Merge pull request #6371 from jmchilton/remove_bad_log_statement

Remove debug statement when creating certain kinds of collections.

martenson committed:
Merge pull request #6366 from jmchilton/job_log_bug

[18.05] Fix small bug in logging output name in job finish.

mvdbeek committed:
Don't rely on the order of history contents in test_map_over_collection_structured_like

nsoranzo committed:
Add gafa.sqlite datatype

Generated by the GAFA tool:

https://toolshed.g2.bx.psu.edu/view/earlhaminst/gafa/

Can be visualised with the Aequatus.js plugin (PR forthcoming).

Work in collaboration with @anilthanki as part of the GeneSeqToFamily
project:

https://github.com/TGAC/earlham-galaxytools/tree/master/workflows/GeneSeqToFamily

dannon committed:
Merge pull request #6372 from nsoranzo/add_gafa.sqlite_datatype

Add gafa.sqlite datatype

dannon committed:
Merge pull request #6367 from jmchilton/finish_opt

Optimizations for dynamic output file collection.

martenson committed:
move the sanitization to the try/except block

as it may fail (and it did, on Main)

natefoo committed:
Merge pull request #6377 from martenson/catch-sanitization-faults

[18.05] Catch sanitization faults

machinelit committed:
Add better data files presentation to isa-tab datatype

guerler committed:
Comma-separate tool names in missing tools error message

ilveroluca commented on lib/galaxy/datatypes/isa.py:
```diff
@@ -448,26 +445,36 @@ def display_data(self, trans, dataset, preview=False, filename=None, to_ext=None
-                        html += '</ul>'
-
-            html += '</body></html>'
+                        file_types = set(x.label for x in assay.data_files)
```
If there's a chance that `assay.data_files` could ever get relatively large, it would be worthwhile to implement this rendering a bit differently.  See the notes below

ilveroluca commented on lib/galaxy/datatypes/isa.py:
```diff
@@ -448,26 +445,36 @@ def display_data(self, trans, dataset, preview=False, filename=None, to_ext=None
-            html += '</body></html>'
+                        file_types = set(x.label for x in assay.data_files)
+                        data_files = {}
+                        for file_type in file_types:
```
This grouping could be done by sorting the entire list by `(label, filename)``and then grouping with `itertools.groupby`

ilveroluca commented on lib/galaxy/datatypes/isa.py:
```diff
@@ -448,26 +445,36 @@ def display_data(self, trans, dataset, preview=False, filename=None, to_ext=None
+                            data_files[file_type] = sorted([x.filename for x in
+                                                     assay.data_files if
+                                                     x.label == file_type])
+                        for label, filenames in data_files.items():
```
use an iterator (`iteritems()`) instead of creating a new list with `items()`

djcomlab replied:
Sure I can update this. It's worth noting that Python 3 `items()` returns an iterator and there is no `iteritems()`, so whenever Galaxy moves to py3 this will be changed back again...

ilveroluca replied:
Indeed, this is an issue.  Given what you reasonably expect this code to see as ISA datasets, it might not be worth changing.  Alternatively, there are some idioms that work well with both Python 2 and 3.  For instance, there's one using `six`, which Galaxy already uses:

```
from six import iteritems
for (key, value) in iteritems(heights):
    ...
```

I pulled that tip from this page :-)  http://python-future.org/compatible_idioms.html

machinelit committed:
Changes as per review comments

machinelit committed:
Fix wrong iterator

djcomlab commented:
Hmm doesn't seem to render now, and it's being a pain to debug so just going to park this as WIP for the moment.

dannon committed:
Merge pull request #6378 from guerler/commaseperror

[18.05] Comma-separate tool names in missing tools error message

jmchilton committed:
Merge pull request #6195 from guerler/shared_data_selector

Common dataset selector

jmchilton committed:
Remove copy-paste mistakes.

dannon committed:
Revert "Remove bam to bai converter"

This reverts commit 7e2727bc0a74868198e7e457f99df3d1e39f1ddb.

jxtx committed:
Some style cleanup.

- Split the Galaxy components of the styleguide into multiple documents
  to make things more manageable and get a better table of contents. Add
  some additional examples for masthead and panels.

- Use flexbox for layout of panels and masthead. This eliminates a lot
  of absolute positioning from the panels and allows the masthead and
  panel headers to layour correctly regardles of height.

- Unify the set of gray levels we are using (to those from Bootstrap 4)

- Add some additional SCSS variables and (small amount of) general
  cleanup to styles.

mvdbeek committed:
Add basic .bai datatype

martenson committed:
Merge pull request #6385 from dannon/revert-baiconverter-removal

[18.01] Revert "Remove bam to bai converter"

jmchilton committed:
Another typo fix.

qiagu committed:
change a file read to use with statement

dannon committed:
Merge pull request #6388 from qiagu/data_column_1

change a file read to use with statement

dannon committed:
Merge pull request #6370 from jmchilton/collection_operations_opt

More correct, more efficient collection operations.

guerler committed:
Position masthead/scratchbook sub icons

nsoranzo committed:
Merge branch 'release_18.01' into release_18.05

nsoranzo committed:
Merge branch 'release_18.05' into dev

jxtx committed:
Merge pull request #6 from guerler/ui-flex-scratchbook

Fixes the position of small sub-icons in masthead

dannon committed:
Merge pull request #6386 from jxtx/ui-flex

Some style cleanup.

Commit:
Fixing <conditional> XML example

dannon committed:
Merge pull request #6399 from almahmoud/dev

Fixing <conditional> XML example

mvdbeek committed:
Allow resource selection for data managers

We used to restrict this to standard tools,
but there's no reason why this shouldn't work
for data manager tools. Fixes https://github.com/galaxyproject/galaxy/issues/6396

martenson committed:
Merge pull request #6402 from mvdbeek/resource_selection_data_manager_tools

Allow resource selection for data managers

machinelit committed:
Add stubs for isatab-viewer plugin

machinelit committed:
Add stubs for isatab-viewer plugin

machinelit committed:
Merge branch 'feature/isa_data_type_plus_enhancements' of https://github.com/phnmnl/galaxy into feature/isa_data_type_plus_enhancements

machinelit committed:
Update isatab-viewer plugin

machinelit committed:
Import plugin resources from previous ISA-Tab visualization work

machinelit committed:
Work towards getting ISA-Tab Viewer to work

machinelit committed:
Implement loading from rawdatadict output from isa-tab datatype

machinelit committed:
Update isa-rwval version in requirements

machinelit committed:
Add function to get raw data out of isa-tab hda as a dictionary

djcomlab commented:
Actually the rendering problem was just a local problem for me running Galaxy.

pkrog commented:
Hi @djcomlab , I was testing it on MTBLS2 and I got this error:
```
127.0.0.1 - - [30/Jun/2018:08:22:03 +0200] "GET /datasets/a660be1fe0057dfb/display/?preview=True HTTP/1.1" 500 - "http://localhost:8080/" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10.13; rv:60.0) Gecko/20100101 Firefox/60.0"
Error - <type 'exceptions.UnicodeDecodeError'>: 'ascii' codec can't decode byte 0xe2 in position 447: ordinal not in range(128)
URL: http://localhost:8080/datasets/a660be1fe0057dfb/display/?preview=True
File '/home/pierrick/dev/galaxy/lib/galaxy/web/framework/middleware/error.py', line 154 in __call__
  app_iter = self.application(environ, sr_checker)
File '/home/pierrick/dev/galaxy/.venv/local/lib/python2.7/site-packages/paste/recursive.py', line 85 in __call__
  return self.application(environ, start_response)
File '/home/pierrick/dev/galaxy/.venv/local/lib/python2.7/site-packages/paste/httpexceptions.py', line 640 in __call__
  return self.application(environ, start_response)
File '/home/pierrick/dev/galaxy/lib/galaxy/web/framework/base.py', line 136 in __call__
  return self.handle_request(environ, start_response)
File '/home/pierrick/dev/galaxy/lib/galaxy/web/framework/base.py', line 215 in handle_request
  body = method(trans, **kwargs)
File '/home/pierrick/dev/galaxy/lib/galaxy/webapps/galaxy/controllers/dataset.py', line 249 in display
  return data.datatype.display_data(trans, data, preview, filename, to_ext, offset=offset, ck_size=ck_size, **kwd)
File '/home/pierrick/dev/galaxy/lib/galaxy/datatypes/isa.py', line 472 in display_data
  return html.encode('utf-8')
UnicodeDecodeError: 'ascii' codec can't decode byte 0xe2 in position 447: ordinal not in range(128)
```

pkrog commented:
Same type of error on MTBLS3 (`'ascii' codec can't decode byte 0xce in position 154: ordinal not in range(128)`), but it works on MTBLS291 and MTBLS4.

djcomlab commented:
@pkrog Yes you're right - seems to be a problem because I removed the use of the HTML sanitizer to get the HTML5 tag `<summary>` to work. Looks like the Galaxy implementation of `sanitize_html()` removes it, perhaps incorrectly. I can try make the appropriate change.

machinelit committed:
Fix sanitize_html() to allow <summary> tags

machinelit committed:
Fixes to rendering HTML preview; add function to get out data for viz plugin

machinelit committed:
Fix erroneous sanitization and fix data files rendering

djcomlab commented:
@pkrog Can you do a quick check on this? If it's alright then we can merge, and then the changes in https://github.com/phnmnl/container-galaxy-k8s-runtime/pull/243 should work correctly.

djcomlab commented:
Closing in favour of https://github.com/phnmnl/container-galaxy-k8s-runtime/pull/244

djcomlab commented:
Whoops closed wrong PR! Reopening

pkrog commented:
Hi @djcomlab, I've tested all four studies MTBLS2, MTBLS3, MTBLS4 and MTBLS291, and it worked perfectly. It is nice to have the list of data files, however I think this is misleading since my ISA tab dataset did not contain the data files but only the metadata files. Another addition would be to check that each file is present inside the ISA-tab dataset and print a tag `(missing in dataset)` at the right of the filename. What do you think?

djcomlab commented:
Sure, sounds good. But if it holds up PR into the release I would not want to wait for further enhancements at this stage. We can always add that in the next RC2.

machinelit committed:
Add display to user of missing datafile in ISA-Tab history dataset

djcomlab commented:
OK, @pkrog I added a quick implementation of your suggestion.

pcm32 commented:
I will revert this merge and merge instead the new branch that is cleaner.
